File: cpp/1107.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int a, c, aux;
    vector<int> heights;
    while (cin >> a >> c) {
        if  (a == 0 && c == 0) {
            break;
        }
        for (int i = 0; i < c; i++) {
            cin >> aux;
            heights.push_back(aux);
        }
        int result = a - heights[0];
        for (int i = 1; i < c; i++) {
            if  (heights[i - 1] > heights[i]) {
                result += heights[i - 1] - heights[i];
            }
        }
        cout << result << '\n';
        heights.clear();
    }
    return 0;
}

File: py/1107.py
def main() -> None:
    while True:
        a, c = [int(i) for i in input().split()]
        if a == c == 0:
            break
        heights = [int(i) for i in input().split()]
        result = a - heights[0]
        for i in range(1, c):
            if heights[i - 1] > heights[i]:
                result += heights[i - 1] - heights[i]
        print(result)


if __name__ == '__main__':
    main()
